clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -49,6 +50,18 @@ object UncommittedGarbageCollector {
     )
     // Read objects from data path (new repository structure)
     var dataDF = new ParallelDataLister().listData(configMapper, dataPath)
+
+    // Save first slice from data prefix
+    if (!dataDF.isEmpty) {
+      // Need the before filter to to exclude slices that are not actually read
```

Comment:
can we extract this implementation from liar objects. it seems until now this method did one thing and now it does two.